Source/Initialization/WarpXInitData.cpp

```diff
@@ -529,3 +531,36 @@ WarpX::InitializeExternalFieldsOnGridUsingParser (
         );
     }
 }
+
+void
+WarpX::PerformanceHints ()
+{
+    // Check requested MPI ranks and available boxes
+    amrex::Long total_nboxes = 0; // on all MPI ranks
+    for (int ilev = 0; ilev <= finestLevel(); ++ilev) {
```

Comment:
The question was raised in Tuesday morning's meeting about whether work is done by level or all levels at once. If it is by level, this should use the number of boxes only on level 0. Correct?

Reply:
This is checking all levels right now, assuming that we can distribute refined levels to otherwise idle ranks that don't have a box from level 0 to crunch on.

Reply:
That was my question, is that a good assumption? Did someone check it?

Reply:
Yes, I think that this is a good assumption, at least *for the starting configuration* (i.e. without explicit load-balancing). 
For instance, using the above example, let us assume that we have 4 MPI rank (numbered from 0 to 3) and 3 boxes on level 0 + 1 box on level 1.
In that case, `amrex` will create the 3 boxes from level 0 on ranks 0,1,2. It will then create the box for level 1, and automatically place it on level 1, because `amrex` uses hints on the memory usage of each MPI ranks, when allocating new boxes.
At least that is my understanding, and I did check this a few months ago. But maybe @WeiqunZhang can confirm.